Feature/incident confidence

austin-carman opened the pull request:
# Description
This pull request adds a confidence rating to the incident card on the admin dashboard under the unapproved, approved, and edit tabs (though the confidence rating appears on editing for reference, it cannot be changed). This has been added to give the admin user more information and confidence about the incident and how closely it relates to the criteria used while scrapping tweets for hits.

Video Walkthrough: 
<a href="https://www.loom.com/share/27aa5328e92844049abd8d9df79ef9d5">
    <p>Feature/incident-confidence - Watch Video</p>
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/27aa5328e92844049abd8d9df79ef9d5-with-play.gif">
  </a>


- [ ] New feature (non-breaking change which adds functionality)

## Change Status

- [ ] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)


# How Has This Been Tested?

- [x] `npm test`

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

austin-carman committed:
added confidence rating for incidents

austin-carman committed:
added confidence rating to edit form - read only

austin-carman committed:
removing logs

imjeremiah commented on src/components/AdminDashboard/CompleteIncident.js:
```diff
@@ -139,20 +139,30 @@ const CompleteIncident = props => {
+              <div>{incident.tags ? incident.tags.join(', ') : ''}</div>
+            </div>
+
+            {/* Confidence Rating: */}
```
Since the confidence rating rates the Force Rank classification only, would it make more sense to put this inline with, or directly under Force Rank? And should we consider possibly relabeling it to something like "Force Rank Confidence" to provide clarity on what it's actually rating?

austin-carman replied:
Good point. I'm going to change the name to "Force Rank Confidence" so that the name is more descriptive and also move it just below the force rank.

Ethan-Edmond reviewed (approved):
Your code does what you set out to do. It doesn't look like it introduces any bugs. An area for improvement I see is that there are a couple of inconsistent formatting changes, and you don't explain those in your video or description. The confidence rating is probably better off thought of in relation to the force rating, but one of me, you or Kelly are going to have to rectify that later in our upcoming merge conflict anyways. I like the idea of keeping the confidence rating displayed while editing and I wouldn't have thought of that. Approved.

austin-carman commented:
The latest commit has made the following changes after peer review:
1. Changed the name from "Confidence Rating" to "Force Rank Confidence"
2. Changed the percentage to round down to a whole number instead of a displaying as a decimal
3. Moved the confidence to just below the "Force Rank" since the confidence refers to this data.

Ethan, could you expound on the inconsistent formatting? Are the inconsistencies you found in the two code blocks labeled, 
{/* Confidence Rating: */}, or elsewhere in the file (as I did notice some inconsistencies in other parts of the file but figured that was outside of the scope of this PR and can fix these inconsistencies in another one)?

Ethan-Edmond commented on src/components/AdminDashboard/CompleteIncident.js:
```diff
@@ -92,17 +90,17 @@ const CompleteIncident = props => {
-                Title:
-              </p>
+              <p className="complete-incident-dropdown-titles-bold">Title:</p>
               <p>{incident.title || '(none)'}</p>
```
Sure I can show you what I mean by inconsistencies, one example would be on lines 93-95 you separated the p element into three lines while on line 103 you collapsed a similar p element to one line. this happens a couple times.

austin-carman replied:
Ah, ok gotcha. Yeah, most of this file was already built out. All I really did was add in the two blocks labeled "Confidence Rating". I'm planning to go through in a separate PR to correct some of these inconsistencies that are found in several files and do some cleaning up with that. More than happy to include that into this PR though if you think that would be better. Thought?

Ethan-Edmond replied:
Honestly it's not a problem, the code still works the same. Do you dislike it? Is it hard for you to read? if not, it's probably good for now.

Ethan-Edmond commented on src/components/AdminDashboard/CompleteIncident.js:
```diff
@@ -81,8 +80,7 @@ const CompleteIncident = props => {
       <div className="complete-incident-dropdown">
-
-        {!editing &&
+        {!editing && (
```
You cut a single line of white space here without doing it across the file.

Ethan-Edmond commented on src/components/AdminDashboard/CompleteIncident.js:
```diff
@@ -139,20 +139,30 @@ const CompleteIncident = props => {
+        )}
 
-        {editing &&
+        {editing && (
```
Your addition of a paren here would be a consistent formatting change, since you did that to the other one too.

Ethan-Edmond commented on src/components/AdminDashboard/CompleteIncident.js:
```diff
@@ -38,7 +37,7 @@ const CompleteIncident = props => {
     setFormValues({
       ...incident,
-      tags: incident.tags ? incident.tags.join(", ") : [],
+      tags: incident.tags ? incident.tags.join(', ') : [],
```
The change to single quotes I would say is consistent since you did it across all of the javascript parts

ahmedseragcodes replied:
Interesting I just saw this on another PR and was wondering why. Well, as long as it works.

ahmedseragcodes commented on src/components/AdminDashboard/CompleteIncident.js:
```diff
@@ -247,7 +257,9 @@ const CompleteIncident = props => {
             <label className="label">
               Tags
-              <span style={{ fontWeight: "normal" }}>&nbsp;(comma separated values)</span>
+              <span style={{ fontWeight: 'normal' }}>
```
I'm a little confused by what the previous code was doing here... specifically with (comma separated values) in a span?

kcjerrell replied:
That was when I added the tags field to the completeincident form. It makes the label look like this:

**Tags** (comma separated values)

(Tags is bold and the rest is not)

ahmedseragcodes reviewed (approved):
Liked the display of functionality in the walkthrough + see no bugs or increased possibility of bugs, approved

imjeremiah reviewed (approved):
Spoke with the author about proposed changes, changes were made. Everything looks good. No merge conflicts. Approved.

kcjerrell reviewed (approved):
This is good. I think every property we store for an incident should be viewable.

paulstgermain reviewed (approved):
Upon discussing the inconsistent formatting with the developer, it turns out this is something stemming from their Linter settings. As such, the formatting of the codebase is something we can take as a separate ticket when we approach shipping for this project.
